### PR TITLE
Composer tutorial: add function and dag

### PR DESCRIPTION
**Pull request by dmitriy-drogovoz**

### CODE REVIEW

**Comment by leahecole** [comment]
Note for other reviewers: This PR is a submission from TVCs that I and the Cloud Composer team have been working with - we've done an extensive review of the Composer-y aspects of the code in their fork and are now looking for final style review - I'm about to add suggestions for region tags @pavel-salnikov

**Review by leahecole — CHANGES_REQUESTED** [review]
1. Please move the function and its test to the `composer/functions` directory - they can be renamed `pubsub_publisher.py` and `pubsub_publisher_test.py` instead of `main.py` and `main_test.py`
2. Add `google-cloud-pubsub==2.13.11` to `composer/functions/requirements.txt` - I think that `google-cloud` isn't needed (I don't see it imported anywhere, but if I'm wrong, let me know)
3. Resolve test failures in the function test - there are NameErrors about `pubsub_function` not being defined. To replicate how the CICD runs the tests locally, copy [`noxfile-template.py`](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/noxfile-template.py) to `composer/functions` (once you've moved the code there) and run `nox -s py-3.8`. See the [Authoring guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md) for more info
4. Resolve lint failures - to replicate the lint session locally, once you have that noxfile from 3 in your directory (you can do the same for the `composer/workflows` directory) you can run `nox -s lint` 

Once this is done, I will add commit suggestions in the code that add "region tags" - comments that will enable @pavel-salnikov to embed the code in the docs

**Comment by leahecole** [comment]
I have triggered the lint job re-run in fusion and it's queued

File: composer/functions/pubsub_publisher.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" This Cloud Function example creates Pub/Sub messages.
+
+    Usage: Replace <PROJECT_ID> with the project ID of your project.
+"""
+
```
Comment (leahecole) [inline composer/functions/pubsub_publisher.py:18]:
```suggestion

# [START composer_pubsub_publisher_function]
```

File: composer/functions/pubsub_publisher.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    # The required data format is a bytestring
+    data = data_str.encode("utf-8")
+    # When you publish a message, the client returns a future.
+    message_length = len(data_str)
+    future = publisher.publish(topic_path,
+                               data,
+                               message_length=str(message_length))
+    print(future.result())
+
+    return f"Message {data} with message_length {message_length} published to {topic_path}."
```
Comment (leahecole) [inline composer/functions/pubsub_publisher.py:57]:
```suggestion
    return f"Message {data} with message_length {message_length} published to {topic_path}."
# [END composer_pubsub_publisher_function]
```

File: composer/workflows/pubsub_trigger_response_dag.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Two Airflow DAGs that demonstrate the mechanism of triggering DAGs with Pub/Sub messages
+
+    Usage: Replace <PROJECT_ID> with the project ID of your project
+"""
+
```
Comment (leahecole) [inline composer/workflows/pubsub_trigger_response_dag.py:18]:
```suggestion

# [START composer_pubsub_trigger_dag]
```

File: composer/workflows/pubsub_trigger_response_dag.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        'target_dag',
+        start_date=datetime(2022, 1, 1),
+        # Not scheduled, trigger only
+        schedule_interval=None,
+        catchup=False) as target_dag:
+
+    some_heavy_task = PythonOperator(task_id='some_heavy_task',
+                                     python_callable=_some_heavy_task)
+
+    (some_heavy_task)
```
Comment (leahecole) [inline composer/workflows/pubsub_trigger_response_dag.py:100]:
```suggestion
    (some_heavy_task)
# [END composer_pubsub_trigger_dag]
```

**Review by leahecole — CHANGES_REQUESTED** [review]
1. Please remove the zip file - users can download the dag and function separately without them needing to be zipped together
2. Please accept the suggestions which will add the region tags for @pavel-salnikov

**Review by leahecole — APPROVED** [review]
@pavel-salnikov PTAL 😄

**Review by pavel-salnikov — APPROVED** [review]

**Comment by leahecole** [comment]
I manually retriggered lint and 3.8

**Pull request merged**